### DISCUSSION TITLE
How to know the current time step in material kernel

### DISCUSSION THREAD

**Discussion by alexchen4ai** [idaholab/moose#19179] [Q&A General] [answered]
Hi Moose experts, I am currently doing a time dependent problem and I wish to implement the  backward Euler integration by AD differentiation.Is it possible that we can get the value of the current time step in material subroutine?

**Comment by hugary1995** [comment, marked as answer]
Yes, you can use `_t` for time and `_t_step` for time step.

**Reply by alexchen4ai** [reply]
Thank you! Could you help explain more? time and time step are from executioner. How can I get the value of _t and _t_step in the material kernel?

**Reply by jiangwen84** [reply]
Either in Material or Kernel,  you can directly use _t and _t_step.  They are member variables declared in the base class.

**Reply by alexchen4ai** [reply]
Thank you! That makes sense!
